File: iris-core/tests/fixtures/symbols/sample.go
package sample

// MaxRetries is the maximum retry count.
const MaxRetries = 3

// AppConfig holds application configuration.
type AppConfig struct {
	Name  string
	Debug bool
}

// Serializable defines a serialization interface.
type Serializable interface {
	Serialize() string
}

// Greet returns a greeting for the given name.
func Greet(name string) string {
	return "Hello, " + name + "!"
}

// NewAppConfig creates a new AppConfig with defaults.
func NewAppConfig() *AppConfig {
	return &AppConfig{
		Name:  "",
		Debug: false,
	}
}

File: iris-core/tests/fixtures/symbols/sample.rb
# Maximum retry count.
MAX_RETRIES = 3

# Configuration for the application.
class AppConfig
  attr_accessor :name, :debug

  def initialize(name, debug: false)
    @name = name
    @debug = debug
  end

  # Check if debug mode is on.
  def debug?
    @debug
  end
end

# A module for serialization.
module Serializable
  def serialize
    to_s
  end
end

# Greet a user by name.
def greet(name)
  "Hello, #{name}!"
end
